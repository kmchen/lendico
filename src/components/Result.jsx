import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Container from '@material-ui/core/Container';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';

const useStyles = makeStyles(theme => ({
  cardGrid: {
    paddingTop: theme.spacing(8),
    paddingBottom: theme.spacing(8),
  },
  card: {
    height: '100%',
    display: 'flex',
    flexDirection: 'column',
  },
  cardContent: {
    flexGrow: 1,
  },
}));

export default function Result({ monthlyInstallment }) {
  const classes = useStyles();
  const result = `Monthly Installment: ${monthlyInstallment}`;
  return (
    <Container className={classes.cardGrid} maxWidth="sm">
      <Card className={classes.card}>
        <CardContent className={classes.cardContent}>
          <Typography>{result}</Typography>
        </CardContent>
      </Card>
    </Container>
  );
}

/* eslint-disable react/forbid-prop-types */
Result.propTypes = {
  monthlyInstallment: PropTypes.string.isRequired,
};
