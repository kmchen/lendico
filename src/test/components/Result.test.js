import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Result from '../../components/Result';

configure({ adapter: new Adapter() });

describe('Results component', () => {
  let ResultsComponent;

  beforeEach(() => {
    const props = {
      monthlyInstallment: '2'
    };
    ResultsComponent = shallow(<Result {...props} />);
  });

  it('should render Results component', () => {
    expect(ResultsComponent).toMatchSnapshot();
  });

  it('should render Card component', () => {
    const cardComponent = ResultsComponent.find('WithStyles(ForwardRef(Card))');
    expect(cardComponent.exists()).toBeTruthy();
  });
});

