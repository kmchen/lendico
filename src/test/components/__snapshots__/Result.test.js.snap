// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results component should render Results component 1`] = `
<WithStyles(ForwardRef(Container))
  className="makeStyles-cardGrid-1"
  maxWidth="sm"
>
  <WithStyles(ForwardRef(Card))
    className="makeStyles-card-2"
  >
    <WithStyles(ForwardRef(CardContent))
      className="makeStyles-cardContent-3"
    >
      <WithStyles(ForwardRef(Typography))>
        Monthly Installment: 2
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(Card))>
</WithStyles(ForwardRef(Container))>
`;
